### DISCUSSION TITLE
laf 云开发简单私有化部署(HTTPS)【转】

### DISCUSSION THREAD

**Discussion by maslow** [labring/laf#176] [Feedbacks]
> 转自：
https://cccy.notion.site/LAF-HTTPS-dc74d97917834991a4fc7f2a930265ed

**Comment by jveay** [comment]
上面的文档，很详细，给了很大的帮助。根据上面的文档还有大佬的帮忙解答。在此延申几个比较小白的问题，这些问题仅仅供docker-compose本地部署https参考。
1.如果是部署在公网.env 文件的域名如何填写以及域名解析如何设置。
DEPLOY_DOMAIN=xxxx.com
SYS_CLIENT_HOST=console.xxxx.com
OSS_DOMAIN=oss.xxxx.com
域名解析设置为：*.xxxx.com
2.根据上面的教程部署后可能会导致oss的没有https，所以需要改docker-compose.yml的配置之后在启动容器
![image](https://user-images.githubusercontent.com/21017524/178246221-85024b68-437f-4641-9eb5-749592826d79.png)
3.使用acme.sh申请泛解析证书命令（下面的所有域名都需替换为使用的域名）
acme.sh --issue --dns dns_ali -d lafyun.com -d *.lafyun.com -d *.oss.lafyun.com
使用该命令会在对应的用户目录下面的/.acme.sh/lafyun.com生成证书，使用里面的lafyun.com.key和fullchain.cer改为使用的名称并且移到cert的文件夹修改nginx的配置后使用下面命令安装加载证书
acme.sh --install-cert -d *.xxx.com --key-file   /cert/tls.key  --fullchain-file /cert/tls.cer--reloadcmd  "docker exec -it $(docker ps | grep 0.0.0.0:443 | awk '{print $1}') nginx -s reload"

**Comment by shmilyin** [comment]
我这边配置完HTTPS后一切正常，只有网站托管设置CNMAE后访问502，不知道你那边是不是这种情况？

**Reply by maslow** [reply]
这个不是问题， 当前的 docker 版部署的这块还没支持 自定义域名， 稍等几天 新的 gateway 发布后就可以了
